**Learning Goals**
- Wrangle data 
- Suset data (Filter)
- Create and use Data Dictionary
- Export Data in CSV Format

# 01. Importing libraries

In [3]:
import pandas as pd
import numpy as np 
import os

In [4]:
# create a string folderpath
folderpath = r'C:\Users\Bamidele\Desktop\Training_content\March 2025 - Data Analytics Cohort\Python\Istacart Basket Project'

In [5]:
# import the dataset as a dataframe using the folder os.path.join
df_ords = pd.read_csv(os.path.join(folderpath, '02 Data', 'Original Data', 'orders.csv'), index_col= False )
df_prods = pd.read_csv(os.path.join(folderpath, '02 Data', 'Original Data', 'products.csv'), index_col= False )

In [6]:
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [7]:
df_ords.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [8]:
# account_NPL - before 2009
# - account_NPL_New - After 2009

**Data wrangling procedures**
- Dropping column
- Renaming Column
- Changing data types
- Transposing data

# Dropping Columns

In [11]:
# df.drop(columns = ['Variable'])

In [12]:
# Dropping eval_set column from orders.csv
df_ords.drop(columns= ['eval_set'])

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0         2539329        1             1          2                  8   
1         2398795        1             2          3                  7   
2          473747        1             3          3                 12   
3         2254736        1             4          4                  7   
4          431534        1             5          4                 15   
...           ...      ...           ...        ...                ...   
3421078   2266710   206209            10          5                 18   
3421079   1854736   206209            11          4                 10   
3421080    626363   206209            12          1                 12   
3421081   2977660   206209            13          1                 12   
3421082    272231   206209            14          6                 14   

         days_since_prior_order  
0                           NaN  
1                          15.0  
2                          21.0  
3                          29.0  
4                          28.0  
...                         ...  
3421078                    29.0  
3421079                    30.0  
3421080                    18.0  
3421081                     7.0  
3421082                    30.0  

[3421083 rows x 6 columns]

In [13]:
df_ords.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [14]:
# Dropping eval_set column from orders.csv
df_ords_2 =  df_ords.drop(columns= ['eval_set'])

In [15]:
# Dropping eval_set column from orders.csv
df_ords.drop(columns= ['eval_set'], inplace= True)

In [16]:
df_ords_2.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [17]:
df_ords.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [18]:
# for missing values, investigate
# df['variable'].value_counts(dropna = False)
df_ords['days_since_prior_order'].value_counts(dropna= False)

days_since_prior_order
30.0    369323
7.0     320608
6.0     240013
4.0     221696
3.0     217005
5.0     214503
NaN     206209
2.0     193206
8.0     181717
1.0     145247
9.0     118188
14.0    100230
10.0     95186
13.0     83214
11.0     80970
12.0     76146
0.0      67755
15.0     66579
16.0     46941
21.0     45470
17.0     39245
20.0     38527
18.0     35881
19.0     34384
22.0     32012
28.0     26777
23.0     23885
27.0     22013
24.0     20712
25.0     19234
29.0     19191
26.0     19016
Name: count, dtype: int64

# Renaming Columns

In [20]:
# Rename the column ordr_dow to orders_day_of_week 
df_ords.rename( columns = {'order_dow': 'orders_day_of_week' }, inplace= True )

In [21]:
df_ords.head(1)

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   

   days_since_prior_order  
0                     NaN

# Changing variable data types

In [23]:
df_ords.describe()

order_id       user_id  order_number  orders_day_of_week  \
count  3.421083e+06  3.421083e+06  3.421083e+06        3.421083e+06   
mean   1.710542e+06  1.029782e+05  1.715486e+01        2.776219e+00   
std    9.875817e+05  5.953372e+04  1.773316e+01        2.046829e+00   
min    1.000000e+00  1.000000e+00  1.000000e+00        0.000000e+00   
25%    8.552715e+05  5.139400e+04  5.000000e+00        1.000000e+00   
50%    1.710542e+06  1.026890e+05  1.100000e+01        3.000000e+00   
75%    2.565812e+06  1.543850e+05  2.300000e+01        5.000000e+00   
max    3.421083e+06  2.062090e+05  1.000000e+02        6.000000e+00   

       order_hour_of_day  days_since_prior_order  
count       3.421083e+06            3.214874e+06  
mean        1.345202e+01            1.111484e+01  
std         4.226088e+00            9.206737e+00  
min         0.000000e+00            0.000000e+00  
25%         1.000000e+01            4.000000e+00  
50%         1.300000e+01            7.000000e+00  
75%         1.600000e+01            1.500000e+01  
max         2.300000e+01            3.000000e+01

In [24]:
# astype

In [25]:
# change the type of a column from int to str
df_ords['order_id'] = df_ords['order_id'].astype('str') 

In [26]:
# check the new type
df_ords['order_id'].dtype

dtype('O')

In [27]:
df_ords.describe()

user_id  order_number  orders_day_of_week  order_hour_of_day  \
count  3.421083e+06  3.421083e+06        3.421083e+06       3.421083e+06   
mean   1.029782e+05  1.715486e+01        2.776219e+00       1.345202e+01   
std    5.953372e+04  1.773316e+01        2.046829e+00       4.226088e+00   
min    1.000000e+00  1.000000e+00        0.000000e+00       0.000000e+00   
25%    5.139400e+04  5.000000e+00        1.000000e+00       1.000000e+01   
50%    1.026890e+05  1.100000e+01        3.000000e+00       1.300000e+01   
75%    1.543850e+05  2.300000e+01        5.000000e+00       1.600000e+01   
max    2.062090e+05  1.000000e+02        6.000000e+00       2.300000e+01   

       days_since_prior_order  
count            3.214874e+06  
mean             1.111484e+01  
std              9.206737e+00  
min              0.000000e+00  
25%              4.000000e+00  
50%              7.000000e+00  
75%              1.500000e+01  
max              3.000000e+01

# Transposing Data

In [29]:
# import the dataset as a dataframe using the folder os.path.join (departments.csv)
df_dep = pd.read_csv(os.path.join(folderpath, '02 Data', 'Original Data', 'departments.csv'), index_col= False )

In [30]:
df_dep.head()

department_id       1      2       3        4        5              6  \
0    department  frozen  other  bakery  produce  alcohol  international   

           7     8                9  ...            12      13         14  \
0  beverages  pets  dry goods pasta  ...  meat seafood  pantry  breakfast   

             15          16         17      18      19    20       21  
0  canned goods  dairy eggs  household  babies  snacks  deli  missing  

[1 rows x 22 columns]

In [31]:
# Transpose df_dep
df_dep.T

0
department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

In [32]:
# Transpose df_dep and out it into df_dep_t
df_dep_t = df_dep.T

In [33]:
df_dep_t

0
department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

In [34]:
df_dep_t.reset_index()

index                0
0   department_id       department
1               1           frozen
2               2            other
3               3           bakery
4               4          produce
5               5          alcohol
6               6    international
7               7        beverages
8               8             pets
9               9  dry goods pasta
10             10             bulk
11             11    personal care
12             12     meat seafood
13             13           pantry
14             14        breakfast
15             15     canned goods
16             16       dairy eggs
17             17        household
18             18           babies
19             19           snacks
20             20             deli
21             21          missing

Take the first row of the the df_dep_t for the header
- iloc (number); loc (Text)

In [36]:
new_header = df_dep_t.iloc[0]

In [37]:
new_header

0    department
Name: department_id, dtype: object

In [38]:
df_dep_t_new = df_dep_t[1:]

In [39]:
df_dep_t_new

0
1            frozen
2             other
3            bakery
4           produce
5           alcohol
6     international
7         beverages
8              pets
9   dry goods pasta
10             bulk
11    personal care
12     meat seafood
13           pantry
14        breakfast
15     canned goods
16       dairy eggs
17        household
18           babies
19           snacks
20             deli
21          missing

In [40]:
df_dep_t_new.columns = new_header # this set header row as the df header

In [41]:
df_dep_t_new

department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

# Data Dictionaries

In [43]:
# {}

In [44]:
data_dict = df_dep_t_new.to_dict('index')

In [45]:
data_dict

{'1': {'department': 'frozen'},
 '2': {'department': 'other'},
 '3': {'department': 'bakery'},
 '4': {'department': 'produce'},
 '5': {'department': 'alcohol'},
 '6': {'department': 'international'},
 '7': {'department': 'beverages'},
 '8': {'department': 'pets'},
 '9': {'department': 'dry goods pasta'},
 '10': {'department': 'bulk'},
 '11': {'department': 'personal care'},
 '12': {'department': 'meat seafood'},
 '13': {'department': 'pantry'},
 '14': {'department': 'breakfast'},
 '15': {'department': 'canned goods'},
 '16': {'department': 'dairy eggs'},
 '17': {'department': 'household'},
 '18': {'department': 'babies'},
 '19': {'department': 'snacks'},
 '20': {'department': 'deli'},
 '21': {'department': 'missing'}}

In [46]:
# first five rows of product dataframe
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [47]:
# get the details of department id 19
print(data_dict.get('19'))

{'department': 'snacks'}


# Subsetting

In [49]:
# Create a subset of df_prods dataframe for only snacks dept data

# df_snacks = df_prods[]

In [50]:
df_snacks = df_prods[ df_prods['department_id'] == 19]

In [51]:
df_snacks.head()

product_id                                 product_name  aisle_id  \
0            1                   Chocolate Sandwich Cookies        61   
15          16                Mint Chocolate Flavored Syrup       103   
24          25      Salted Caramel Lean Protein & Fiber Bar         3   
31          32                Nacho Cheese White Bean Chips       107   
40          41  Organic Sourdough Einkorn Crackers Rosemary        78   

    department_id  prices  
0              19     5.8  
15             19     5.2  
24             19     1.9  
31             19     4.9  
40             19     6.5

In [52]:
# using loc to subset
df_snacks_2 = df_prods.loc[ df_prods['department_id']== 19]

In [53]:
df_snacks_2.head()

product_id                                 product_name  aisle_id  \
0            1                   Chocolate Sandwich Cookies        61   
15          16                Mint Chocolate Flavored Syrup       103   
24          25      Salted Caramel Lean Protein & Fiber Bar         3   
31          32                Nacho Cheese White Bean Chips       107   
40          41  Organic Sourdough Einkorn Crackers Rosemary        78   

    department_id  prices  
0              19     5.8  
15             19     5.2  
24             19     1.9  
31             19     4.9  
40             19     6.5

In [54]:
# Using Isin to subset
df_snacks_3 = df_prods.loc[    df_prods['department_id'].isin([19])               ]

In [55]:
df_snacks_3.head()

product_id                                 product_name  aisle_id  \
0            1                   Chocolate Sandwich Cookies        61   
15          16                Mint Chocolate Flavored Syrup       103   
24          25      Salted Caramel Lean Protein & Fiber Bar         3   
31          32                Nacho Cheese White Bean Chips       107   
40          41  Organic Sourdough Einkorn Crackers Rosemary        78   

    department_id  prices  
0              19     5.8  
15             19     5.2  
24             19     1.9  
31             19     4.9  
40             19     6.5

In [56]:
# Using Isin to subset
df_snacks_4 = df_prods.loc[    df_prods['department_id'].isin([17,18,19])               ]

In [57]:
df_snacks_4.tail()

product_id                                   product_name  aisle_id  \
49669       49665  Super Dark Coconut Ash & Banana Chocolate Bar        45   
49670       49666                  Ginger Snaps Snacking Cookies        61   
49672       49668                 Apple Cinnamon Scented Candles       101   
49675       49671                           Milk Chocolate Drops        45   
49683       49679                        Famous Chocolate Wafers        61   

       department_id  prices  
49669             19     6.9  
49670             19     5.2  
49672             17     5.6  
49675             19     3.0  
49683             19     6.0

# Exporting Dataframes

In [59]:
df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0  2539329        1             1                   2                  8   
1  2398795        1             2                   3                  7   
2   473747        1             3                   3                 12   
3  2254736        1             4                   4                  7   
4   431534        1             5                   4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [60]:
# Export your cleaned dataframe to the prepared folder
df_ords.to_csv(os.path.join(folderpath, '02 Data', 'Prepared Data', 'orders_wrangled.csv')  )

In [61]:
# Prepared by Bams

In [119]:
# df_dep_t_new.to_csv(os.path.join(folderpath, '02 Data', 'Prepared Data', 'department_wrangled.csv')  )

In [ ]:
df_dep_t_new.to_pickle(os.path.join(folderpath, '02 Data', 'Prepared Data', 'department_wrangled.pkl')  )

In [ ]:
bamsTest = pd.read_csv(os.path.join(folderpath, '02 Data', 'Prepared Data', 'department_wrangled.csv'), index_col= False )

In [ ]:
bamsTest

In [ ]:
bamsTest2 = pd.read_pickle(os.path.join(folderpath, '02 Data', 'Prepared Data', 'department_wrangled.pkl') )

In [ ]:
bamsTest2

# New Task

**1. Find another identifier variable in the df_ords dataframe that doesn’t need to be included in your analysis as a numeric variable and change it to a suitable format.?**

In [67]:
# first 5 rows
df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0  2539329        1             1                   2                  8   
1  2398795        1             2                   3                  7   
2   473747        1             3                   3                 12   
3  2254736        1             4                   4                  7   
4   431534        1             5                   4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [69]:
# The new identifyer will be user_id, hence, it should be converted to str
df_ords['user_id'] = df_ords['user_id'].astype('str') 

In [79]:
# check the new data type of user_id
df_ords['user_id'].dtype

dtype('O')

**2. Look for a variable in your df_ords dataframe with an unintuitive name and change its name without overwriting the dataframe.**

In [84]:
# Change order_number to user_order_number
df_ords_new = df_ords.rename(columns={'order_number': 'user_order_number'} )

In [88]:
# confirm the name change is working properly
df_ords_new.columns

Index(['order_id', 'user_id', 'user_order_number', 'orders_day_of_week',
       'order_hour_of_day', 'days_since_prior_order'],
      dtype='object')

**3. Your client wants to know what the busiest hour is for placing orders. 
Find the frequency of the corresponding variable and share your findings.**

In [108]:
# calculate the frequency of each hour of day
busy_hour = df_ords['order_hour_of_day'].value_counts()
print(busy_hour)

order_hour_of_day
10    288418
11    284728
15    283639
14    283042
13    277999
12    272841
16    272553
9     257812
17    228795
18    182912
8     178201
19    140569
20    104292
7      91868
21     78109
22     61468
23     40043
6      30529
0      22758
1      12398
5       9569
2       7539
4       5527
3       5474
Name: count, dtype: int64


In [112]:
# identify the busiest hour
busiest_hour = busy_hour.idxmax()
busiest_count = busy_hour.max()
print(f"The busiest hour for placing orders is {busiest_hour}:00 with {busiest_count} orders.")

The busiest hour for placing orders is 10:00 with 288418 orders.


The busiest hour is **10 a.m** with a total order of **288,418**

**4. Determine the meaning behind a value of 4 in the "department_id" column within the df_prods dataframe using a data dictionary.**

In [114]:
# Display the department name corresponding to department_id = 4
print(data_dict.get('4'))

{'department': 'produce'}


**5. The sales team in your client’s organization wants to know more about breakfast item sales. Create a subset containing only the required information.**

In [121]:
# Create a subset for breakfast products
df_breakfast = df_prods[df_prods['department_id'] == 14]

In [123]:
df_breakfast

product_id                                       product_name  \
27             28                                  Wheat Chex Cereal   
33             34                                                NaN   
67             68                            Pancake Mix, Buttermilk   
89             90                                       Smorz Cereal   
210           211   Gluten Free Organic Cereal Coconut Maple Vanilla   
...           ...                                                ...   
49330       49326                            Cereal Variety Fun Pack   
49395       49391            Light and Fluffy Buttermilk Pancake Mix   
49547       49543                          Chocolate Cheerios Cereal   
49637       49633               Shake 'N Pour Buttermilk Pancake Mix   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   

       aisle_id  department_id  prices  
27          121             14    10.1  
33          121             14    12.2  
67          130             14    13.7  
89          121             14     3.9  
210         130             14     3.6  
...         ...            ...     ...  
49330       121             14     9.1  
49395       130             14     2.0  
49547       121             14    10.8  
49637       130             14    14.2  
49667        57             14    10.2  

[1116 rows x 5 columns]

In [127]:
df_breakfast.shape

(1116, 5)

**6. They’d also like to see details about products that customers might use to throw dinner parties. Your task is to find all observations from the entire dataframe that include items from the following departments: alcohol, deli, beverages, and meat/seafood. You’ll need to present this subset to your client.**

In [130]:
# Create the subset
df_dinner_party = df_prods[df_prods['department_id'].isin([5, 7, 12, 20])]

In [132]:
df_dinner_party

product_id                                    product_name  aisle_id  \
2               3            Robust Golden Unsweetened Oolong Tea        94   
6               7                  Pure Coconut Water With Orange        98   
9              10  Sparkling Orange Juice & Prickly Pear Beverage       115   
10             11                               Peach Mango Juice        31   
16             17                               Rendered Duck Fat        35   
...           ...                                             ...       ...   
49676       49672                          Cafe Mocha K-Cup Packs        26   
49679       49675             Cinnamon Dolce Keurig Brewed K Cups        26   
49680       49676                          Ultra Red Energy Drink        64   
49686       49682                              California Limeade        98   
49688       49684       Vodka, Triple Distilled, Twist of Vanilla       124   

       department_id  prices  
2                  7     4.5  
6                  7     4.4  
9                  7     8.4  
10                 7     2.8  
16                12    17.1  
...              ...     ...  
49676              7     6.5  
49679              7    14.0  
49680              7    14.5  
49686              7     4.3  
49688              5     5.3  

[7650 rows x 5 columns]

In [134]:
df_dinner_party.shape

(7650, 5)

**7. It’s important that you keep track of total counts in your dataframes. How many rows does the last dataframe you created have?**

In [142]:
df_dinner_party.shape

(7650, 5)

**8. Someone from the data engineers team in Instacart thinks they’ve spotted something strange about the customer with a "user_id" of “1.” Extract all the information you can about this user.**

In [145]:
df_ords.head()

order_id user_id  order_number  orders_day_of_week  order_hour_of_day  \
0  2539329       1             1                   2                  8   
1  2398795       1             2                   3                  7   
2   473747       1             3                   3                 12   
3  2254736       1             4                   4                  7   
4   431534       1             5                   4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [153]:
# df_prods.head()

In [169]:
user_id_1 = df_ords[ df_ords['user_id'] == '1']

In [171]:
user_id_1

order_id user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329       1             1                   2                  8   
1   2398795       1             2                   3                  7   
2    473747       1             3                   3                 12   
3   2254736       1             4                   4                  7   
4    431534       1             5                   4                 15   
5   3367565       1             6                   2                  7   
6    550135       1             7                   1                  9   
7   3108588       1             8                   1                 14   
8   2295261       1             9                   1                 16   
9   2550362       1            10                   4                  8   
10  1187899       1            11                   4                  8   

    days_since_prior_order  
0                      NaN  
1                     15.0  
2                     21.0  
3                     29.0  
4                     28.0  
5                     19.0  
6                     20.0  
7                     14.0  
8                      0.0  
9                     30.0  
10                    14.0

**9. You also need to provide some details about this user’s behavior. What basic stats can you provide based on the information you have?**

In [173]:
user_id_1.describe()

order_number  orders_day_of_week  order_hour_of_day  \
count     11.000000           11.000000          11.000000   
mean       6.000000            2.636364          10.090909   
std        3.316625            1.286291           3.477198   
min        1.000000            1.000000           7.000000   
25%        3.500000            1.500000           7.500000   
50%        6.000000            3.000000           8.000000   
75%        8.500000            4.000000          13.000000   
max       11.000000            4.000000          16.000000   

       days_since_prior_order  
count               10.000000  
mean                19.000000  
std                  9.030811  
min                  0.000000  
25%                 14.250000  
50%                 19.500000  
75%                 26.250000  
max                 30.000000

**10. Check the organization and structure of your notebook. Be sure to include section headings and code comments.**

In [185]:
df_ords.shape

(3421083, 6)

In [187]:
df_prods.shape

(49693, 5)

**11. Export your df_ords dataframe as “orders_wrangled.csv” in your “Prepared Data” folder.**

In [192]:
# Export your cleaned dataframe to the prepared folder
df_ords.to_csv(os.path.join(folderpath, '02 Data', 'Prepared Data', 'orders_wrangled.csv')  )

**12. Export the df_dep_t_new dataframe as “departments_wrangled.csv” in your “Prepared Data” folder so that you have a “.csv” file of your departments data in the correct format.**

In [196]:
# export df_dep_t_new to csv
df_dep_t_new.to_csv(os.path.join(folderpath, '02 Data', 'Prepared Data', 'department_wrangled.csv') )

In [ ]:
# export df_dep_t_new to pickle file
df_dep_t_new.to_pickle(os.path.join(folderpath, '02 Data', 'Prepared Data', 'department_wrangled.pkl')  )